lib/Util/TypeBasedHeapCloning.cpp

```diff
@@ -454,8 +455,9 @@ NodeID TypeBasedHeapCloning::cloneObject(NodeID o, const DIType *type, bool reus
         {
             clone = addCloneFIObjNode(fiObj->getMemObj());
         }
-        else if (const DummyObjPN *dummyObj = SVFUtil::dyn_cast<DummyObjPN>(obj))
+        else
```

Comment:
This changes the underlying semantics.

Reply:
May trigger a null pointer at line 451

Reply:
> This changes the underlying semantics.

Can `isa<type>(obj)` be true, and `dyn_cast<type>(obj)` still fail? Otherwise I don't see how this would change anything, as we have only two possible cases, which should be covered by both `if {} else if {}` and `if {} else {}`.

Reply:
make sense.